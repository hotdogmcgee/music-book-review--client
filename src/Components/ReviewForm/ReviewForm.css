.ReviewForm {
  text-align: center;
}

.ReviewForm select,
.ReviewForm label {
  font-size: 1.1em;
}

.select {
  text-align: center;
  top: auto;
  bottom: 100%;
}

.ReviewForm label {
  display: inline-block;
  margin: 10px 10px 20px;
  margin-left: 0;
}

.ReviewForm__textarea {
  width: calc(80vw);
}

@media screen and (min-width: 850px) {
  .ReviewForm__textarea {
    width: calc(60vw);
  }
}
