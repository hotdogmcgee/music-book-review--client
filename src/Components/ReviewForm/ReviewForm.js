import React, { Component } from "react";
import BookContext from "../../Contexts/BookContext";
import BooksApiService from "../../services/books-api-service";
import { Button, Textarea } from "../Utils/Utils";
import TokenService from "../../services/token-service";
import "./ReviewForm.css";

export default class ReviewForm extends Component {
  static defaultProps = {
    onReviewSuccess: () => {},
    onReviewFailure: () => {}
  };
  static contextType = BookContext;

  handleSubmit = ev => {
    ev.preventDefault();
    this.context.clearError();

    if (!TokenService.hasAuthToken()) {
      this.context.setError(
        "You must log in or register before adding a review"
      );
      this.props.onReviewFailure();
      return;
    }
    const { book } = this.context;
    const { text, rating } = ev.target;

    BooksApiService.postReview(book.id, Number(rating.value), text.value)
      .then(thing => this.context.addReview(thing))
      .then(() => {
        text.value = "";
        this.props.onReviewSuccess();
      })
      .catch(this.context.setError("You must be logged in to submit a review"));
  };

  render() {
    return (
      <>
        <form className="ReviewForm" onSubmit={this.handleSubmit}>
          <div className="text">
            <Textarea
              required
              aria-label="Type a review..."
              name="text"
              id="text"
              cols="20"
              rows="5"
              placeholder="Type a review.."
              className="ReviewForm__textarea"
            ></Textarea>
          </div>

          <div className="select">
            <label htmlFor="rating">Rate this book!</label>
            <select
              required
              aria-label="Rate this thing!"
              name="rating"
              id="rating"
            >
              <option value="1">1 Star</option>
              <option value="2">2 Stars</option>
              <option value="3">3 Stars</option>
              <option value="4">4 Stars</option>
              <option value="5">5 Stars</option>
            </select>
          </div>

          <Button type="submit">Post review</Button>
        </form>
      </>
    );
  }
}
