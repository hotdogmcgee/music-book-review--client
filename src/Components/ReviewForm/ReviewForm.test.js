import React from 'react';
import ReviewForm from './ReviewForm'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('ReviewForm component', () => {

    it('renders a ReviewForm by default', () => {
        const wrapper = shallow(<ReviewForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})