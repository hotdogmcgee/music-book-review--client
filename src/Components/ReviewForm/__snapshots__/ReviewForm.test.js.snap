// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm component renders a ReviewForm by default 1`] = `
<Fragment>
  <form
    className="ReviewForm"
    onSubmit={[Function]}
  >
    <div
      className="text"
    >
      <Textarea
        aria-label="Type a review..."
        className="ReviewForm__textarea"
        cols="20"
        id="text"
        name="text"
        placeholder="Type a review.."
        required={true}
        rows="5"
      />
    </div>
    <div
      className="select"
    >
      <label
        htmlFor="rating"
      >
        Rate this book!
      </label>
      <select
        aria-label="Rate this thing!"
        id="rating"
        name="rating"
        required={true}
      >
        <option
          value="1"
        >
          1 Star
        </option>
        <option
          value="2"
        >
          2 Stars
        </option>
        <option
          value="3"
        >
          3 Stars
        </option>
        <option
          value="4"
        >
          4 Stars
        </option>
        <option
          value="5"
        >
          5 Stars
        </option>
      </select>
    </div>
    <Button
      type="submit"
    >
      Post review
    </Button>
  </form>
</Fragment>
`;
